packages/core/src/clients/middleware/retry.ts

```diff
@@ -102,3 +110,16 @@ const cancellableSleep = (timeoutMs: number, abortSignal?: AbortSignal) => {
 	});
 	return sleepPromise;
 };
+
+const isMetadataBearer = (response: unknown): response is MetadataBearer =>
+	response && typeof response['$metadata'] === 'object';
```

Comment:
Nit: Should be able to use optional chaining
```suggestion
	typeof response?.['$metadata'] === 'object';
```